sips/sip-012/sip-012-cost-limits-network-upgrade.md

```diff
@@ -0,0 +1,674 @@
+# Preamble
+
+SIP Number: 012
+
+Title: Burn height selection for network-upgrade to introduce new cost-limits
+
+Authors:
+* 0xAsteria <asteria@syvita.org>
+* Aaron Blankstein <aaron@hiro.so>
+* Diwaker Gupta <diwaker@hiro.so>
+* Jason Lau <jason@okcoin.com>
+* Ludovic Galabru <ludo@hiro.so>
+* Trevor Owens <trevor@stacks.ac>
+* Xan Ditkoff <xan@daemontechnologies.co>
+
+Consideration: Governance, Technical
+
+Type: Consensus
+
+Status: Draft
+
+Created: 2021-10-08
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+The current Clarity cost limits were set very conservatively in Stacks 2.0: transactions with contract-calls frequently exceed these limits, which negatively affects transaction throughput. This SIP proposes an update to these cost-limits via a network upgrade and further, that the network upgrade be executed at a block height chosen by an off-chain process described in this SIP.
+
+
+# License and Copyright
+
+This SIP is made available under the terms of the Creative Commons CC0 1.0 Universal license, available at https://creativecommons.org/publicdomain/zero/1.0/ This SIP’s copyright is held by the Stacks Open Internet Foundation.
+
+
+# Introduction
+
+Blocks on the Stacks blockchain tend to have anywhere between 10 to 50 transactions per block -- this is lower than the demands of many workloads and also lower than the theoretical maximum one would expect. On the other hand, the mempool consistently has hundreds of valid transactions pending at any given time; at peak there have been several thousand pending transactions. So what is preventing more transactions from being included in blocks?
+
+The answer is cost-limits. An analysis of “full blocks” (meaning blocks where at least one cost dimension exceeds the block limit -- see SIP-006 for a description of all cost dimensions) found that roughly 72% of those blocks hit the `runtime` limit and ~27% hit the `read_count` limit. Another analysis of all transactions rejected due to cost limits found that 90% of those transactions exceeded the `runtime` limit. So the `runtime` limits are the primary bottleneck right now.
```

Comment:
Can you add the spreadsheet and graph as a supplemental item to this SIP?  See SIP-000 for details on how to do this.